packages/webpack/react-webpack-plugin/src/loaders/background.ts

```diff
@@ -5,50 +5,44 @@ import { createRequire } from 'node:module';
 
 import type { LoaderContext } from '@rspack/core';
 
-import { getBackgroundTransformOptions } from './options.js';
-import type { ReactLoaderOptions } from './options.js';
+import { getBackgroundTransformOptions } from './loader-options.js';
+import type { ReactLoaderOptions } from './loader-options.js';
 
-function backgroundLoader(
+async function backgroundLoader(
   this: LoaderContext<ReactLoaderOptions>,
   content: string,
-): void {
+): Promise<void> {
+  const callback = this.async();
   const require = createRequire(import.meta.url);
-  const { transformPath = '@lynx-js/react/transform' } = this.getOptions();
-  const { transformReactLynxSync } = require(
-    transformPath,
+  const { transformReactLynx } = require(
+    '@lynx-js/react/transform',
   ) as typeof import('@lynx-js/react/transform');
-  const result = transformReactLynxSync(
+  const result = await transformReactLynx(
     content,
     getBackgroundTransformOptions.call(this),
   );
 
   if (result.errors.length > 0) {
     for (const error of result.errors) {
       if (this.experiments?.emitDiagnostic) {
-        // Rspack with `emitDiagnostic` API
         try {
           this.experiments.emitDiagnostic({
```

Comment:
I don't think we still need to use `experiments.emitDiagnostic` when there are no location.